packages/kit/src/runtime/server/page/load_node.js

```diff
@@ -300,12 +335,148 @@ export async function load_node({
 		return;
 	}
 
+	// generate __data.json files when prerendering
+	if (shadow.body && state.prerender) {
+		const pathname = `${event.url.pathname}/__data.json`;
+
+		const dependency = {
+			response: new Response(undefined),
+			body: JSON.stringify(shadow.body)
+		};
+
+		state.prerender.dependencies.set(pathname, dependency);
+	}
+
 	return {
 		node,
+		props: shadow.body,
 		loaded: normalize(loaded),
 		stuff: loaded.stuff || stuff,
 		fetched,
 		set_cookie_headers,
 		uses_credentials
 	};
 }
+
+/**
+ *
+ * @param {import('types/internal').SSRPage} route
+ * @param {import('types/hooks').RequestEvent} event
+ * @param {boolean} prerender
+ * @returns {Promise<import('types/endpoint').ShadowData>}
+ */
+async function load_shadow_data(route, event, prerender) {
+	if (!route.shadow) return {};
+
+	try {
+		const mod = await route.shadow();
+
+		if (prerender && (mod.post || mod.put || mod.del || mod.patch)) {
+			throw new Error('Cannot prerender pages that have shadow endpoints with mutative methods');
+		}
+
+		const method = event.request.method.toLowerCase().replace('delete', 'del');
+		const handler = mod[method];
+
+		if (!handler) {
+			// TODO figure out what to do here. is 405 Method Not Allowed
```

Comment:
"TODO figure out what to do here" always makes me nervous. What are the variables? What might we be falling through to?

Reply:
in a situation like this...

```
src/routes/specific.js
src/routes/specific.svelte
src/routes/[unspecific].js
src/routes/[unspecific].svelte
```

...where you `POST` to `/specific`, one of two things could happen if `src/routes/specific.js` doesn't export `post`: we could throw a 405, or we could fall through to `src/routes/[unspecific].js` to see if that's able to fulfil the request. But that means responding with a 404 rather than a 405 if it's unable to do so.

Reply:
Hm. I think a 405 makes the most sense, then. I don't see a ton of value in letting shadow endpoints implicitly fall through to other endpoints. It would also make apps harder to reason about.

I assume that if `src/routes/specific.js` had a `post` that explicitly returned `{ fallthrough: true }` that that fallthrough would be handled the same way any other would? If so, I'm find with disallowing implicit fallthrough here.

Reply:
cool — removed the comment